### PR TITLE
Docs organize user setting profiles

### PR DESCRIPTION
**Pull request by DanRoscigno**
This adds an overview to the settings section of the docs explaining the difference between global server settings and query-level settings and provides examples of the different ways of setting the query-level settings.

### Changelog category (leave one):
- Documentation (changelog entry is not required)

Related to ClickHouse/ClickHouse#112119 , I will tackle that one next (it should be very similar)

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit 7756620f15bded229086a2a60593a94bb6f7d236 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52041/7756620f15bded229086a2a60593a94bb6f7d236/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docs/en/operations/server-configuration-parameters/settings.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ---
 slug: /en/operations/server-configuration-parameters/settings
 sidebar_position: 57
```
Comment (rschu1ze) [inline docs/en/operations/server-configuration-parameters/settings.md:3]: This is the current order of doc pages:

image

In my view, a more logical order would be:
1. Settings Overview
2. Configuration files
3. User Settings
4. Settings Profiles
5. Constraints on Settings
6. Global Server Settings
7. Merge Tree Settings
8. Core Settings
9. Format Settings
10. Restrictions on Query Complexity

Pages "Permissions for Queries" and "Memory Overcommit" look out-of-place and is ideally integrated into an existing page.

File: docs/en/operations/settings/index.md [hunk 2]
```diff
@@ -7,90 +7,16 @@ pagination_next: en/operations/settings/settings
 
 # Settings Overview
 
-There are multiple ways to define ClickHouse settings. Settings are configured in layers, and each subsequent layer redefines the previous values of a setting.
+There are two main groups of ClickHouse settings:
 
-The order of priority for defining a setting is:
+- Global server settings
```
Comment (rschu1ze) [inline docs/en/operations/settings/index.md:12]: There are two main groups of ClickHouse settings:
- Global server settings
- Session and query-level settings

File: docs/en/operations/settings/index.md [hunk 3]
```diff
@@ -7,90 +7,16 @@ pagination_next: en/operations/settings/settings
...
+- Query-level settings
 
-1. Settings in the `users.xml` server configuration file
+The main distinction between global server settings and query-level settings is that
```
Comment (rschu1ze) [inline docs/en/operations/settings/index.md:15]: The main distinction between global server settings and session / query-level settings is that global server settings must be set in configuration files while session / query-level settings can be set in configuration files or with SQL queries.

File: docs/en/operations/settings/settings-query-level.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
+---
+sidebar_label: Query-level Settings
```
Comment (rschu1ze) [inline docs/en/operations/settings/settings-query-level.md:2]: Let's rename this to include "session-level" settings.

**Review by rschu1ze — APPROVED** [review]

**Comment by rschu1ze** [comment]
@DanRoscigno The auto-merger submitted this PR a bit too early into the codebase :-) Kindly see my remaining comments (they are not critical).

**Pull request merged**